src/test/java/tests/conformance/AbstractConformanceTest.java

```diff
@@ -263,225 +222,169 @@ static Mode fromEnvironment() {
+    private ExpectedFact(String commentText) {
+      this.commentText = commentText;
+    }
 
-      /** The comment text representing this expected fact. */
-      public String commentText() {
-        return commentText;
-      }
+    private final String commentText;
 
-      /** Returns {@code true} if this is a legacy {@code jspecify_nullness_mismatch} assertion. */
-      public boolean isNullnessMismatch() {
-        return NULLNESS_MISMATCH.matcher(commentText).matches();
-      }
+    /** The comment text representing this expected fact. */
+    public String commentText() {
+      return commentText;
+    }
 
-      @Override
-      public boolean equals(Object obj) {
-        if (obj == this) {
-          return true;
-        }
-        if (!(obj instanceof ExpectedFact)) {
-          return false;
-        }
-        ExpectedFact that = (ExpectedFact) obj;
-        return this.commentText.equals(that.commentText);
-      }
+    /** Returns {@code true} if this is a legacy {@code jspecify_nullness_mismatch} assertion. */
+    public boolean isNullnessMismatch() {
+      return NULLNESS_MISMATCH.matcher(commentText).matches();
+    }
 
-      @Override
-      public int hashCode() {
-        return commentText.hashCode();
+    @Override
+    public boolean equals(Object obj) {
+      if (obj == this) {
+        return true;
       }
-
-      @Override
-      public String toString() {
-        return commentText;
+      if (!(obj instanceof ExpectedFact)) {
+        return false;
       }
+      ExpectedFact that = (ExpectedFact) obj;
+      return this.commentText.equals(that.commentText);
     }
 
-    /**
-     * An assertion that the tool behaves in a way consistent with a specific fact about a line in
-     * the source code. Some of these facts indicate that according to the JSpecify specification,
-     * the code in question may have an error that should be reported to users; other expected facts
-     * are informational, such as the expected nullness-augmented type of an expression.
-     */
-    public static final class ExpectedFactAssertion extends ConformanceTestAssertion {
-      private final long lineNumber;
-      private final ExpectedFact fact;
-
-      ExpectedFactAssertion(Path file, long lineNumber, ExpectedFact fact) {
-        super(file);
-        this.lineNumber = lineNumber;
-        this.fact = fact;
-      }
+    @Override
+    public int hashCode() {
+      return commentText.hashCode();
+    }
 
-      /** Returns the line number of the code in the source file to which this assertion applies. */
-      public long getLineNumber() {
-        return lineNumber;
-      }
+    @Override
+    public String toString() {
+      return commentText;
+    }
+  }
 
-      /** Returns the fact expected at the file and line number. */
-      public ExpectedFact getFact() {
-        return fact;
-      }
+  /**
+   * An assertion that the tool behaves in a way consistent with a specific fact about a line in the
```

Comment:
The relationship between `Fact`, `ExpectedFact`, and `ExpectedFactAssertion` isn't clear at first glance. That's partially about class naming (`ExpectedFact` is not a fact, but `ExpectedFactAssertion` is) and partially about how `ExpectedFact` and `ExpectedFactAssertion` have very similar Javadoc.

Reply:
Yeah. I want to replace the current `ExpectedFact` with a string, and then rename `ExpectedFactAssertion` to `ExpectedFact` in a followup PR. Should I do that in this PR instead?

Reply:
Followup is fine, thanks.